### ISSUE TITLE
malsd_batch beam search decodes labels for a batch row with encoder_output_length == 0

### ISSUE THREAD

**Issue by udsy19** [NVIDIA-NeMo/NeMo#16161]
**Describe the bug**

`malsd_batch` beam search decodes labels for a batch row whose `encoder_output_length` is 0.

`nemo/collections/asr/parts/submodules/rnnt_malsd_batched_computer.py:513` builds the initial
`active_mask` with an `encoder_output_length > 0` term, under the comment *"In mixed batches some rows
may have `encoder_output_length == 0`"*. The two in-loop recomputations drop it (`:731` eager, `:1415`
CUDA-graph). Since `last_timesteps` is `clamp_min(encoder_output_length - 1, 0)` it is `0`, not `-1`,
for such a row, so `time_indices <= last_timesteps` is `0 <= 0` and the row is re-activated on the
first update, emitting up to `max_symbols_per_step` labels.

This is a regression from #15753, which added the `clamp_min` and compensated only at the two init
sites. Before it `last_timesteps` was `-1` there and every recomputation was correctly `False`.

Batched streaming inference produces this shape: in
`examples/asr/asr_chunked_inference/rnnt/speech_to_text_streaming_infer_rnnt.py:474`, once a row's
audio is consumed its `out_len` is `0` while the rest of the batch keeps decoding.

**Steps/Code to reproduce bug**

```python
import torch
from nemo.collections.asr.modules import RNNTDecoder, RNNTJoint
from nemo.collections.asr.parts.submodules.rnnt_beam_decoding import BeamBatchedRNNTInfer

V = ["a", "b", "c", "d", "e", "f"]
torch.manual_seed(1234)
dec = RNNTDecoder({'pred_hidden': 8, 'pred_rnn_layers': 1}, len(V) + 1).eval()
joint = RNNTJoint(
    {'encoder_hidden': 8, 'pred_hidden': 8, 'joint_hidden': 8, 'activation': 'relu'},
    len(V) + 1, vocabulary=V,
).eval()
bs = BeamBatchedRNNTInfer(
    dec, joint, blank_index=len(V), beam_size=2, search_type="malsd_batch",
    max_symbols_per_step=5, allow_cuda_graphs=False,
)
torch.manual_seed(0)
with torch.no_grad():
    hyps = bs(encoder_output=torch.randn(2, 8, 9) * 3.0,
              encoded_lengths=torch.tensor([9, 0], dtype=torch.int32))[0]
print(repr(hyps[1].y_sequence))
```

On `6281c939aa` this prints `array([4, 4, 4, 4, 4])`. On `a409dfd1bc` (the commit before #15753) it
prints `array([], dtype=int64)`. `maes_batch`, TDT `malsd_batch` and `greedy_batch` also print the
empty sequence.

**Expected behavior**

A row with `encoded_lengths == 0` decodes nothing.

**Environment overview**

- Environment location: bare metal
- Method of NeMo install: from source, `pip install -e .`

**Environment details**

- Ubuntu 24.04 · PyTorch 2.13.0+cpu · Python 3.12.3
